### PR TITLE
Extended UniRec API: ur_time_from_str()

### PR DESCRIPTION
**Pull request by cejkato2**
Add function to convert string into ur_time_t (moved and revisited existing code).

Add unit test that covers ur_time_t.

### CODE REVIEW

File: unirec/unirec.c [hunk 1]
```diff
@@ -1350,6 +1336,35 @@ int ur_set_from_string(const ur_template_t *tmpl, void *data, ur_field_id_t f_id
    return rv;
 }
 
+uint8_t ur_time_from_string(ur_time_t *ur, const char *str)
```
Comment (jaroslavh) [inline unirec/unirec.c]: We should check whether *str is not NULL. It would result in a Segmentation fault as strptime does not check it. This should be also tested in tests.

File: unirec/tests/test_time.c [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * in contract, strict liability, or tort (including negligence or
+ * otherwise) arising in any way out of the use of this software, even
+ * if advised of the possibility of such damage.
+ *
+ */
+
+#include "../ur_time.h"
+#include <stdio.h>
+
+int main()
```
Comment (jaroslavh) [inline unirec/tests/test_time.c]: Add tests for empty string.

File: unirec/unirec.c [hunk 3]
```diff
@@ -1350,6 +1336,35 @@ int ur_set_from_string(const ur_template_t *tmpl, void *data, ur_field_id_t f_id
...
+            msec = atoi(res + 1);
+         }
+         *ur = ur_time_from_sec_msec((uint64_t) sec, msec);
+      } else {
+         goto failed_time_parsing;
+      }
+      /* success */
+      return 0;
+   } else {
+failed_time_parsing:
```
[the remark this answers is not in the document]
Comment (cejkato2) [reply unirec/unirec.c]: No.
It is used in two "branches" where an error can occur. (either `timegm()` fails or `strptime()` fails -> both results in retval 1)
